cmd/libsnap-confine-private/apparmor-support.c

```diff
@@ -46,28 +46,34 @@ void sc_init_apparmor_support(struct sc_apparmor *apparmor)
 	// Use aa_is_enabled() to see if apparmor is available in the kernel and
 	// enabled at boot time. If it isn't log a diagnostic message and assume
 	// we're not confined.
-	if (aa_is_enabled() != true) {
+	if (aa_is_enabled() == 0) {
 		switch (errno) {
 		case ENOSYS:
 			debug
 			    ("apparmor extensions to the system are not available");
 			break;
+		case EBUSY:
+			debug
+			    ("apparmor is enabled but the interface is private");
+			break;
 		case ECANCELED:
 			debug
 			    ("apparmor is available on the system but has been disabled at boot");
 			break;
 		case EPERM:
 			// NOTE: fall-through
 		case EACCES:
-			debug
-			    ("insufficient permissions to determine if apparmor is enabled");
+			die("insufficient permissions to determine if apparmor is enabled");
 			// since snap-confine is setuid root this should
 			// never happen so likely someone is trying to
 			// manipulate our execution environment - fail hard
-
-			// fall-through
+			break;
 		case ENOENT:
+			die("apparmor is enabled but the interface is not available");
+			break;
 		case ENOMEM:
+			die("insufficient memory to determine if apparmor is available");
```

Comment:
not sure this one is worthy of a separate message, unless die() never allocates anything inside